File: src/main/java/medium/ReverseWordsInString.java
package medium;

public class ReverseWordsInString {
  public String solve(String s) {
    StringBuilder sb = new StringBuilder();
    String[] words = s.trim().split("\\s+");
    int right = words.length - 1;
    while (right >= 0) {
      String word = words[right];
      if (!word.isBlank() || !word.isEmpty() || !word.equals(" ")) {
        sb.append(word).append(" ");
      }
      right--;
    }
    return sb.toString().trim();
  }
}

File: lib/medium/reverse_words_in_string.dart
String reverseWordsInString(String s) {
  List<String> words = s.trim().split(RegExp(r'\s+'));
  List<String> result = [];
  for (int i = words.length - 1; i >= 0; i--) {
    result.add(words[i]);
  }
  return result.join(" ");
}
